### PR TITLE
[ogre-next] update to v2.3.0

### LINKED ISSUE #13214: [OGRE3D] Update to the v2.2.3
Hi,
OGRE3D v2.2.3 has been released since july 5 2020, a vcpkg search shows that the one available dates back from 2019-10-20
Best,

### PR DESCRIPTION
**Pull request by FrankXie05**
Fix #13214
Fix #12647
[ogre-next] update to the latest version 2.3.0
All features have been tested successfullly in the following triplet:

- x64-windows
- x86-windows

### CODE REVIEW

File: ports/ogre-next/portfile.cmake [hunk 1]
```diff
@@ -28,19 +27,12 @@ else()
...
 
-vcpkg_configure_cmake(
-    SOURCE_PATH ${SOURCE_PATH}
-    PREFER_NINJA
+vcpkg_cmake_configure(
+    SOURCE_PATH "${SOURCE_PATH}"
     OPTIONS
         -DOGRE_BUILD_DEPENDENCIES=OFF
-        -DOGRE_BUILD_SAMPLES=OFF
+        -DOGRE_BUILD_SAMPLES=ON
```
Comment (JackBoosY) [inline ports/ogre-next/portfile.cmake:35]: This should be set as `OFF`.

File: ports/ogre-next/portfile.cmake [hunk 2]
```diff
@@ -50,72 +42,40 @@ vcpkg_configure_cmake(
         -DOGRE_INSTALL_PDB=OFF
         -DOGRE_INSTALL_SAMPLES=OFF
         -DOGRE_INSTALL_TOOLS=OFF
-        -DOGRE_INSTALL_CMAKE=ON
         -DOGRE_INSTALL_VSPROPS=OFF
         -DOGRE_STATIC=${OGRE_STATIC}
-        -DOGRE_CONFIG_THREAD_PROVIDER=std
+        -DOGRE_CONFIG_THREAD_PROVIDER=0
```
Comment (JackBoosY) [inline ports/ogre-next/portfile.cmake:47]: Why change this?

File: ports/ogre-next/portfile.cmake [hunk 3]
```diff
@@ -50,72 +42,40 @@ vcpkg_configure_cmake(
...
         -DOGRE_BUILD_RENDERSYSTEM_D3D11=ON
         -DOGRE_BUILD_RENDERSYSTEM_GL=ON
         -DOGRE_BUILD_RENDERSYSTEM_GL3PLUS=ON
         -DOGRE_BUILD_RENDERSYSTEM_GLES=OFF
         -DOGRE_BUILD_RENDERSYSTEM_GLES2=OFF
-# Optional stuff
-        ${FEATURE_OPTIONS}
-# vcpkg specific stuff
+        -DOGRE_BUILD_PLUGIN_PFX=ON
```
Comment (JackBoosY) [inline ports/ogre-next/portfile.cmake:53]: Why add this?

File: ports/ogre-next/portfile.cmake [hunk 4]
```diff
@@ -50,72 +42,40 @@ vcpkg_configure_cmake(
...
         -DOGRE_CMAKE_DIR=share/ogre-next
-)
-
-vcpkg_install_cmake()
-vcpkg_fixup_cmake_targets()
-
-file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/share)
-file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/include)
+		)
```
Comment (JackBoosY) [inline ports/ogre-next/portfile.cmake:55]:
```suggestion
)

```

**Review by JackBoosY — CHANGES_REQUESTED** [review]

File: ports/ogre-next/toolchain_fixes.patch [hunk 5]
```diff
@@ -18,41 +18,7 @@ index a7a3796..2e21403 100644
...
-+if(NOT DEFINED OGRE_CMAKE_DIR)
-+  if(WIN32 OR APPLE)
-+    set(OGRE_CMAKE_DIR "CMake")
-+  else()
-+    set(OGRE_CMAKE_DIR "${OGRE_LIB_DIRECTORY}/OGRE/cmake")
-+  endif()
- endif()
- configure_package_config_file(${OGRE_TEMPLATES_DIR}/OGREConfig.cmake.in ${PROJECT_BINARY_DIR}/cmake/OGREConfig.cmake
-     INSTALL_DESTINATION ${OGRE_CMAKE_DIR}
-diff --git a/CMake/Packages/FindCg.cmake b/CMake/Packages/FindCg.cmake
```
Comment (FrankXie05) [inline ports/ogre-next/toolchain_fixes.patch:43]: The new version deletes this file: `FindCg.cmake`

File: ports/ogre-next/toolchain_fixes.patch [hunk 6]
```diff
@@ -18,41 +18,7 @@ index a7a3796..2e21403 100644
...
--  set(OGRE_CMAKE_DIR "${OGRE_LIB_DIRECTORY}/OGRE/cmake")
-+if(NOT DEFINED OGRE_CMAKE_DIR)
-+  if(WIN32 OR APPLE)
-+    set(OGRE_CMAKE_DIR "CMake")
-+  else()
-+    set(OGRE_CMAKE_DIR "${OGRE_LIB_DIRECTORY}/OGRE/cmake")
-+  endif()
- endif()
- configure_package_config_file(${OGRE_TEMPLATES_DIR}/OGREConfig.cmake.in ${PROJECT_BINARY_DIR}/cmake/OGREConfig.cmake
-     INSTALL_DESTINATION ${OGRE_CMAKE_DIR}
```
Comment (FrankXie05) [inline ports/ogre-next/toolchain_fixes.patch:42]: The new version merged this modification

File: ports/ogre-next/portfile.cmake [hunk 7]
```diff
@@ -59,63 +58,31 @@ vcpkg_configure_cmake(
...
+  file(REMOVE "${REL_CFGS}")
 endif()
 
-file(GLOB DBG_CFGS ${CURRENT_PACKAGES_DIR}/debug/bin/*.cfg)
+file(GLOB DBG_CFGS "${CURRENT_PACKAGES_DIR}/debug/bin/*.cfg")
 if(DBG_CFGS)
-  file(COPY ${DBG_CFGS} DESTINATION ${CURRENT_PACKAGES_DIR}/debug/lib)
-  file(REMOVE ${DBG_CFGS})
+  file(COPY "${DBG_CFGS}" DESTINATION "${CURRENT_PACKAGES_DIR}/debug/lib")
+  file(REMOVE "${DBG_CFGS}")
```
Comment (JackBoosY) [inline ports/ogre-next/portfile.cmake:75]:
```suggestion
  file(COPY ${DBG_CFGS} DESTINATION "${CURRENT_PACKAGES_DIR}/debug/lib")
  file(REMOVE ${DBG_CFGS})
```
`DBG_CFGS` is a list, not a path.

File: ports/ogre-next/portfile.cmake [hunk 8]
```diff
@@ -59,63 +58,31 @@ vcpkg_configure_cmake(
...
-file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/share)
-file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/include)
-
-file(GLOB REL_CFGS ${CURRENT_PACKAGES_DIR}/bin/*.cfg)
+file(GLOB REL_CFGS "${CURRENT_PACKAGES_DIR}/bin/*.cfg")
 if(REL_CFGS)
-  file(COPY ${REL_CFGS} DESTINATION ${CURRENT_PACKAGES_DIR}/lib)
-  file(REMOVE ${REL_CFGS})
+  file(COPY "${REL_CFGS}" DESTINATION "${CURRENT_PACKAGES_DIR}/lib")
+  file(REMOVE "${REL_CFGS}")
```
Comment (JackBoosY) [inline ports/ogre-next/portfile.cmake:69]:
```suggestion
  file(COPY ${REL_CFGS} DESTINATION "${CURRENT_PACKAGES_DIR}/lib")
  file(REMOVE ${REL_CFGS})
```

File: ports/ogre-next/portfile.cmake [hunk 9]
```diff
@@ -59,63 +58,31 @@ vcpkg_configure_cmake(
...
+  file(REMOVE "${DBG_CFGS}")
 endif()
 
 if(VCPKG_LIBRARY_LINKAGE STREQUAL "static")
-    file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/bin ${CURRENT_PACKAGES_DIR}/debug/bin)
-endif()
-
-#Remove OgreMain*.lib from lib/ folder, because autolink would complain, since it defines a main symbol
-#manual-link subfolder is here to the rescue!
-if(VCPKG_TARGET_IS_WINDOWS)
```
Comment (JackBoosY) [inline ports/ogre-next/portfile.cmake:92]: Why remove this? Are `OgreMain` and `OgreMainStatic` not exist?

**Review by JackBoosY — CHANGES_REQUESTED** [review]

File: ports/ogre-next/portfile.cmake [hunk 10]
```diff
@@ -28,18 +26,26 @@ else()
     set(OGRE_STATIC OFF)
 endif()
 
+if (("openvr" IN_LIST FEATURES) AND (VCPKG_LIBRARY_LINKAGE STREQUAL static))
```
Comment (JackBoosY) [inline ports/ogre-next/portfile.cmake:29]:
```suggestion
if (("openvr" IN_LIST FEATURES) AND (VCPKG_LIBRARY_LINKAGE STREQUAL "static"))
```

File: ports/ogre-next/portfile.cmake [hunk 11]
```diff
@@ -28,18 +26,26 @@ else()
...
+    set (CMAKE_DISABLE_FIND_PACKAGE_OPENVR ON)
+    message (" OpenVR only supports dynamic library linkage. Building dynamic library.")
```
Comment (JackBoosY) [inline ports/ogre-next/portfile.cmake:31]: I don't understand this.

**Review by JackBoosY — CHANGES_REQUESTED** [review]

File: ports/ogre-next/portfile.cmake [hunk 12]
```diff
@@ -28,18 +26,26 @@ else()
     set(OGRE_STATIC OFF)
 endif()
 
+if (("openvr" IN_LIST FEATURES) AND (VCPKG_LIBRARY_LINKAGE STREQUAL "static"))
+    set (CMAKE_DISABLE_FIND_PACKAGE_OPENVR ON)
+    message (" OpenVR not support static build.")
+endif()
```
Comment (JackBoosY) [inline ports/ogre-next/portfile.cmake:32]: According to
https://github.com/microsoft/vcpkg/blob/4f778cf094a65f836167eb301bdf62b7ba632c68/ports/openvr/vcpkg.json#L7
And
https://github.com/microsoft/vcpkg/blob/4f778cf094a65f836167eb301bdf62b7ba632c68/ports/openvr/portfile.cmake#L1
openvr can be built on static-windows triplet. So I assume ogre-next can find and use it correctly.

**Review by JackBoosY — CHANGES_REQUESTED** [review]

**Review by JackBoosY — APPROVED** [review]

**Pull request merged**